Source/NavierStokes.cpp

```diff
@@ -339,6 +339,9 @@ NavierStokes::advance (Real time,
         MultiFab& S_old = get_old_data(State_Type);
 	// NOTE have_divu is now a static var in NSBase
         mac_project(time,dt,S_old,&mac_rhs,umac_n_grow,true);
+    } else { 
+	u_mac[0].FillBoundary(0, 1, geom.periodicity());
+        u_mac[1].FillBoundary(0, 1, geom.periodicity());
```

Comment:
You need to loop over AMREX_SPACEDIM in case you are in 3D

Reply:
yes, l am trying the other way around using create_umac_grown(umac_n_grow). I will update in a minute.